src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -327,7 +342,9 @@ - (void)invokeCall {
     }
   }];
   // Now that the RPC has been initiated, request writes can start.
-  [_requestWriter startWithWriteable:self];
+  @synchronized(_requestWriter) {
```

Comment:
As described above, all access to the _requestWrite's state, both getting and setting, have to be _synchronized on the same object. so @synchronized(self){ }